omod/src/main/java/org/openmrs/module/locationbasedaccess/fragment/controller/field/LocationsFragmentController.java

```diff
@@ -0,0 +1,36 @@
+package org.openmrs.module.locationbasedaccess.fragment.controller.field;
+
+import org.openmrs.Location;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.appui.UiSessionContext;
+import org.openmrs.ui.framework.fragment.FragmentModel;
+import javax.servlet.http.HttpSession;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Fields and actions to provide registrationInfo view.
+ *
+ */
+public class LocationsFragmentController {
+
+    public void controller(FragmentModel model,
+                           HttpSession session) {
+        String locationSessionAttribute = UiSessionContext.LOCATION_SESSION_ATTRIBUTE;
+        Integer sessionLocationId = (Integer) session.getAttribute(locationSessionAttribute);
+
+        List<Location> activeLocations =new ArrayList<Location>();
+        activeLocations.add(Context.getLocationService().getLocation(sessionLocationId));
```

Comment:
Doesn't this logic of selecting the active location as default, belong to the view layer?

Reply:
To change in the view layer, I need to implement through the AngularJS, Other than this I followed this logic to set that value as the first value. So it will be automatically selected :smile:

Reply:
Would there be any problem with that?

Reply:
I just skipped that one for now and thought to implement later. It will be an immediate solution for now. Otherwise, We can remove this for now, and add later with Angular. Is it alright?

Reply:
I want this to have as little code as possible.

Reply:
then, I have removed this part for now